From e61a4af8bcca71e1d24f92b09344715c878ebe59 Mon Sep 17 00:00:00 2001
From: "Peter A. Bigot" <pab@pabigot.com>
Date: Wed, 14 Aug 2013 17:12:49 -0500
Subject: [PATCH 31/31] OMAP2/3 clock: fix sprz319 erratum 2.1

The original patch misinterprets the tables; the N value is the divide value
minus one, while the M value is correct as given.

Upstream-Status: Pending

Signed-off-by: Peter A. Bigot <pab@pabigot.com>
---
 arch/arm/mach-omap2/clock3xxx.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap2/clock3xxx.c b/arch/arm/mach-omap2/clock3xxx.c
index 6b611e1..600e218 100644
--- a/arch/arm/mach-omap2/clock3xxx.c
+++ b/arch/arm/mach-omap2/clock3xxx.c
@@ -52,12 +52,12 @@ static int omap3_dpll5_apply_erratum21(struct clk *clk, struct clk *dpll5_m2)
 	int i, rv;
 	static const struct dpll_settings precomputed[] = {
 		/* From DM3730 errata (sprz319e), table 36
-		* +1 is because the values in the table are register values;
-		* dpll_program() will subtract one from what we give it,
+		* N+1 is because the values in the table are register values;
+		* dpll_program() will subtract one from the N we give it,
 		* so ...
 		*/
-		{ 13000000, 443+1, 5+1, 8 },
-		{ 26000000, 443+1, 11+1, 8 }
+		{ 13000000, 443, 5+1, 8 },
+		{ 26000000, 443, 11+1, 8 }
 	};
 
 	sys_clk = clk_get(NULL, "sys_ck");
-- 
1.7.9.5

